src/tests/ftest/util/agent_utils.py

```diff
@@ -33,91 +33,64 @@
 
-class NodeListType:
-    # pylint: disable=too-few-public-methods
-    """
-    Simple enum to represent all possible node types. To be expanded if
-    needed.
-    """
-    SERVER = 0
-    CLIENT = 1
 
-def node_setup_okay(node_list, node_server_type):
-    """
-    Verify the required environment for each node is present.  For now this
-    is defined to mean the domain socket directory is present.  May expand
-    this later.
+class AgentFailed(Exception):
+    """Agent didn't start/stop properly."""
+
 
-    node_list - The set of DAOS nodes for which to verify environment. Node
-                lists are expected to contain exclusively agent or
-                exclusively server nodes; they may not be mixed.
-    """
-    if node_server_type == NodeListType.SERVER:
-        socket_dir = "/var/run/daos_server"
-    elif node_server_type == NodeListType.CLIENT:
-        socket_dir = "/var/run/daos_agent"
-    else:
-        raise AgentFailed("Unknown node type, exiting")
-
-    okay = True
-    failed_node = None
-    for node in node_list:
-        cmd = "test -d " + socket_dir
-        resp = subprocess.call(["ssh", node, cmd])
-        if resp != 0:
-            okay = False
-            failed_node = node
-            break
-    return okay, failed_node, socket_dir
-
-# Disabling check for this function as in this case, more variables is more
-# clear than encapsulating in dict or object
 def run_agent(basepath, server_list, client_list=None):
-    """
-    Makes sure the environment is setup for the security agent and then launches
+    """Start daos agents on the specified hosts.
+
+    Make sure the environment is setup for the security agent and then launches
     it on the compute nodes.
 
     This is temporary; presuming the agent will deamonize at somepoint and
     can be started killed more appropriately.
 
-    basepath    --root directory for DAOS repo or installation
-    client_list --those nodes that are acting as compute nodes in the test
-    server_list --those nodes that are acting as server nodes in the test
+    Args:
+        basepath (str): root directory for DAOS repo or installation
+        server_list (list): nodes acting as server nodes in the test
+        client_list (list, optional): nodes acting as client nodes in the test.
+            Defaults to None.
+
+    Raises:
+        AgentFailed: if there is an error starting the daos agents
+
+    Returns:
+        dict: set of subprocess sessions
+
     """
     sessions = {}
-
     user = getpass.getuser()
 
-    retcode, node, agent_dir = node_setup_okay(server_list, NodeListType.SERVER)
-    if not retcode:
-        raise AgentFailed("Server node " + node + " does not have directory "
-                          + agent_dir + " set up correctly for user "
-                          + user + ".")
-
     # if empty client list, 'self' is effectively client
     if client_list is None:
         client_list = [socket.gethostname().split('.', 1)[0]]
 
```

Comment:
This assumption is an issue on A21 cluster not in our env;  but it brings up a question on portability.    Why do we assume that that the testnode/VM has daos installed?

Reply:
Because it's the job of whatever runs the tests to ensure that whatever is being tested is already installed.  The tests' job is to run tests, not prepare nodes for the test.

This is why in our CI pipeline, we have processes that install whatever is going to be tested before we call the test running process.

Reply:
Hrm.  In rereading your question, is your question actually "why is the testnode/VM being used as a daos client at all and not just being the test runner only?"

Reply:
It feels like what is being said here is all tests should explicitly specify at least one client and one server and therefore if line 70 should be reporting an error.  Tests will then be more portable.  This sort of change would benefit from #746 (due to the `-tc` laumch.py option) and should either be part of that PR, or better yet, a new one.

Reply:
So, yes.  If what @mjean308 is asking is why is the test runner also a daos client (for many tests), I would generally agree that it's best to isolate the code being tested onto a node other than the node that is running the tests.

I'm not sure that's as easy as just defining a client in all tests' YAML files thoufgh.  Many tests have been written to use the Python DAOS API directly so those tests would need to be converted to a style (perhaps some kind of RPC mechanism) where that Python DAOS API client is on a remote node and being remotely controlled by the Avocado test.

There is some support in Avocado for running tests on remote nodes.  I have not investigated exactly what that is but perhaps that could be of benefit here too.

Additionally, there is a slight bit of economics in running the client on the test-runner as it means that all other available (8 in the case of our CI clusters) nodes can be DAOS servers.  Although that could easily be ameliorated by just adding a VM or two more to the hardware clusters to be clients as all of the benefits of hardware nodes relate to servers I think.

Reply:
> I'm not sure that's as easy as just defining a client in all tests' YAML files though.

Yes, I agree and I did not mean to imply it would be simple.  In my mind that's probably the strongest argument for a unique PR.

> Additionally, there is a slight bit of economics in running the client on the test-runner

Agreed.  The additional VMs would probably be the best option, but nothing would stop the yaml file from specifying the launch node as the client.